dask/async.py

```diff
@@ -422,8 +422,10 @@ def get_async(apply_async, num_workers, dsk, result, cache=None,
 
     if isinstance(result, list):
         result_flat = set(flatten(result))
-    else:
+    elif ishashable(result) and result in dsk:
         result_flat = set([result])
+    else:
+        return result
```

Comment:
Raising a key error on a bad key might be possible, but it's something that we would need to investigate in a bit more depth to see if anything else depends on this behavior.